src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,55 @@
 package core.basesyntax.impl;
 
+import core.basesyntax.Pair;
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    public static final int STORAGE_SIZE = 10;
+    public static final int NO_RESULT = -1;
+    private final Pair<K, V>[] storage;
+    private int size;
+
+    public StorageImpl() {
+        storage = new Pair[STORAGE_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        Pair<K, V> toSave = new Pair<>(key, value);
+        int position = findExistingPositionByKey(key);
+        if (position != NO_RESULT) {
+            storage[position] = toSave;
+        } else {
+            if (size < STORAGE_SIZE) {
+                storage[size++] = toSave;
+            } else {
+                throw new ArrayStoreException("Storage is out of free space!");
+            }
```

Comment:
you can simplify this if-else statment
```suggestion
        } else if (size < STORAGE_SIZE) {
                storage[size++] = toSave;
        } 
            throw new ArrayStoreException("Storage is out of free space!");
        }
```